### PR TITLE
added tests for HW3

### PR DESCRIPTION
**Pull request by ajm2605**
Here is my HW3 tests sorry for the delay

### CODE REVIEW

File: tests/test_load_config.py [hunk 1]
```diff
@@ -17,39 +16,39 @@ def config_file_path():
...
+def test_load_config_smoke(config_file_path):
     """Smoke test for load_config function.
     author: ajm
     reviewer: 
     category: smoke test"""
-    config_path = config_file_path()
 
-    assert config_path is not None, "config path returned None"
-    assert config_path.exists(), f"Config file does not exist at {config_path}"
+    assert config_file_path is not None, "config path returned None"
```
Comment (zach8421) [inline tests/test_load_config.py]: load_config returns {} for an empty YAML file (yaml.safe_load(f) or {}), so this test will fail if the function stays consistent with that behavior. Previously the test was checking for {}, which matched the function output.

File: tests/test_load_config.py [hunk 2]
```diff
@@ -17,39 +16,39 @@ def config_file_path():
...
-    config = load_config(config_path)
+    config = load_config(config_file_path)
 
     assert isinstance(config, dict), "Config is not a dictionary"
     assert 'project_name' in config, "Missing 'project_name' in config"
     assert config['project_name'] == 'humanSayMonkeyDo', "Incorrect 'project_name' value"
 
-def test_load_config_edge_case_empty_file(tmp_path):
-    """Edge case test for load_config with an empty file.
+def test_load_config_edge_case_empty_file():
```
Comment (zach8421) [inline tests/test_load_config.py]: Before this change the test accepted tmp_path, which allowed pytest to generate a temp directory for this test. Dropping tmp_path removes that benefit and ties the test to the repo’s working directory.
